src/lib/utils/thorchain/memo/assertAndProcessMemo.ts

```diff
@@ -82,13 +82,33 @@ export const assertAndProcessMemo = (memo: string): string => {
     case '=':
     case 's': {
       // SWAP:ASSET:DESTADDR:LIM/INTERVAL/QUANTITY:AFFILIATE:FEE
-      const [_action, asset, destAddr, limit, , fee] = memo.split(':')
+      const [
+        _action,
+        asset,
+        destAddr,
+        limit,
+        ,
+        fee,
+        aggregatorLastTwoChars,
+        finalAssetAddressShortened,
+        finalAssetAmountOut,
+      ] = memo.split(':')
 
       assertMemoHasAsset(asset, memo)
       assertMemoHasDestAddr(destAddr, memo)
       assertIsValidLimit(limit, memo)
 
-      return `${_action}:${asset}:${destAddr}:${limit}:${THORCHAIN_AFFILIATE_NAME}:${fee || 0}`
+      const swapOutParts = (() => {
+        if (aggregatorLastTwoChars && finalAssetAddressShortened && finalAssetAmountOut) {
+          return `:${aggregatorLastTwoChars}:${finalAssetAddressShortened}:${finalAssetAmountOut}`
```

Comment:
Add assertions here before returning:

- `assertMemoHasAggregatorAddress` (pretty much a copy paste job from `assertMemoHasDestAddr`)
- `assertMemoHasFinalAssetAddress` (same thing)
- `assertIsValidFinalAssetLimit` (Mostly a copy and paste job from `assertIsValidLimit`, though should obviously handle the ramifications of the exponent magic notation)

Reply:
Done in 03978b18e16